src/autocomplete.js

```diff
@@ -666,8 +666,14 @@ class CompletionProvider {
     
         var matches = [];
         this.completers = editor.completers;
-        var total = editor.completers.length;
-        editor.completers.forEach(function(completer, i) {
+        var matchingCompleters = util.getCompletersMatchingTriggerCharacter(editor);
+        // if there are no matches or there is a prefix, and the prefix is longer than the threshold,
+        // then use all completers
+        if (matchingCompleters.length === 0 || (prefix && prefix.length >= editor.$liveAutocompletionThreshold)) {
```

Comment:
If there is no completer registered for trigger char `{`, and I press `{` does this mean that all registered completers would now get triggered by this non prefix char?

Reply:
no, autocomplete won't be triggered at all if there is no prefix (word characters A-Za-z) or trigger char does not match any of the completers, see https://github.com/InspiredGuy/ace/blob/invoke-only-matching-completers-for-trigger-chars/src/ext/language_tools.js#L160

Reply:
ah right, ty